kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/ResolverAAImpl.kt

```diff
@@ -346,6 +347,26 @@ class ResolverAAImpl(
         }
     }
 
+    @KspExperimental
+    override fun getSubpackagesOf(packageName: String): Sequence<String> {
+        return analyze {
+            fun KtPackageSymbol.subPackages(): Sequence<KtPackageSymbol> = getPackageScope()
+                .getPackageSymbols()
+                .distinct()
```

Comment:
Needing to `distinct()` may indicate some issues in the underlying Analysis API. Have you tried removing `distinct()`? If it works then it works. If not, feel free to leave it there and we'll investigate.